src/core/api/replication_policy.go

```diff
@@ -242,6 +238,48 @@ func (r *ReplicationPolicyAPI) Delete() {
 	}
 }
 
+// the execution's status will not be updated if it is not queried
+// so need to check the status of tasks to determine the status of
+// the execution
+func hasRunningExecutions(policyID int64) (bool, error) {
+	_, executions, err := replication.OperationCtl.ListExecutions(&models.ExecutionQuery{
+		PolicyID: policyID,
+	})
+	if err != nil {
+		return false, err
+	}
+	for _, execution := range executions {
+		if execution.Status != models.ExecutionStatusInProgress {
+			continue
+		}
+		_, tasks, err := replication.OperationCtl.ListTasks(&models.TaskQuery{
+			ExecutionID: execution.ID,
+		})
+		if err != nil {
+			return false, err
+		}
+		for _, task := range tasks {
+			if isTaskRunning(task) {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+func isTaskRunning(task *models.Task) bool {
```

Comment:
please either rename the func or add more comments, as it's not just want to check running only.